cylc/flow/scripts/lint.py

```diff
@@ -490,25 +635,49 @@ def main(parser: COP, options: 'Values', *targets) -> None:
         if not target.exists():
             LOG.warn(f'Path {target} does not exist.')
         else:
+            ruleset_default = False
+            if options.linter == 'all':
+                options.linter = ['728', 'style']
+            elif options.linter == '':
+                options.linter = ['728', 'style']
+                ruleset_default = True
+            else:
+                options.linter = [options.linter]
+            tomlopts = get_pyproject_toml(target)
```

Comment:
This requires the config file to be at the level of the directory you are linting which is fine but currently there is a risk of this file not being picked up e.g. `cylc lint ./sites` (lint a subdir below the flow.cylc).

I think this command should switch from `DIR` to `WORKFLOW | PATH` and use the same workflow ID argument parsing as the offline commands (e.g. validate).

Basically just run the list of `targets` through `parse_ids` e.g:

```python
parse_ids(
    *targets,
    src=True,
    constraint='workflows',
)
```

This way the `cylc lint` command will only run on top-level workflow directories and the config file will always be respected.

Reply:
Done